docs/development/feature_documentation/README.md

```diff
@@ -0,0 +1,12 @@
+# Feature Documentation
```

Comment:
It feels weird to call this "Feature Documentation" when it is under "development"... I don't have a better name unfortunately.

Reply:
Hmm, "Component Documentation" perhaps?

Reply:
"Internal Documentation"? "Design Documentation"?

Reply:
Internal Documentation sounds good!